### ISSUE TITLE
Compiled x265.exe (32bit) can't be stripped, won't run either!

### ISSUE THREAD

**Issue by Vangelis66** [jb-alvarado/media-autobuild_suite#301]
Unfortunately, yet again something goes wrong :-(

Suite is in snapshot https://github.com/jb-alvarado/media-autobuild_suite/commit/8f91f595b1c5a7e69bf3f55906a6c6b8c2671e1e (of March 9th). 
The last time the suite was run (the evening of March 10th), 
everything compiled fine and I was a happy guy...

Yesterday (14/03/2016) I again fired up  the suite; 
a number of msys2/mingw32 updates were found, 
downloaded and successfully installed; I let it run overnight 
and went to sleep. On wake up, there was no minnty console 
window found open, a clear indication the suite exited 
in a normal fashion. No new logs.zip was created either.

I have grown a habit of inspecting "compile.log" regardless 
and it was there I discovered the anomalies:

[1m┌ x265 hg (B[m.............................................. [[33mUpdates found(B[m]
├ Running uninstall...
├ Building 12-bit lib for multilib...
├ Running cmake...
├ Running ninja...
├ Building 10-bit lib for multilib...
├ Running cmake...
├ Running ninja...
├ Building multilib lib/bin...
├ Running cmake...
├ Running ninja...
├ Running install...
G:\Compiler-v2\media-autobuild_suite\msys32\mingw32\bin\strip.exe:G:/Compiler-v2/media-autobuild_suite/local32/bin-video/x265.exe: File format not recognized

[1m└ x265-hg (B[m.................................................... [[34mUpdated(B[m]

Indeed, /local32/bin-video/x265.exe was rather big (7.12MiB), 
so I thought this was just an unstripped binary (the stripped ones 
measure usually 6.30 MiB), but the real nasty surprise came 
when I tried to run said binary:

`G:\Compiler-v2\media-autobuild_suite\local32\bin-video>x265 -V`
`Program too big to fit in memory`

`G:\COMPIL~1\MEDIA-~1\local32\BIN-VI~1>`

it is evident something went awry during compilation; 
as said, no logs.zip was created but I composed one 
myself, which is attached: [20160314_logs.zip](https://github.com/jb-alvarado/media-autobuild_suite/files/173175/20160314_logs.zip)

One other thing that caught my attention was:

`[33mffmpeg-0001-configure-Try-pkg-config-first-with-a-few-libs.patch(B[m
	File not found online. Using local copy.`
but after searching it appears to be related to 
https://github.com/jb-alvarado/media-autobuild_suite/commit/ab288adf5b5f65814d0202067dd2119566a428fc ; ffmpeg,exe binary was compiled fine despite 
and, unlike x265.exe one, it does also run fine!

As another test, prior to filing this issue, I 
deleted "/build/x265-hg", "/local32/bin-video/x265.exe", 
"/local32/include/x265.h (& x265_config.h)", 
"/local32/lib/pkgconfig/x265.pc" and "/local32/lib/libx265.a" 
and opted to recompile only "kvazaar.exe" and "x265.exe". 
Alas, the same miserable outcome with x265.exe!
Manually composed: [20150315_logs.zip](https://github.com/jb-alvarado/media-autobuild_suite/files/173188/20150315_logs.zip)

I am afraid that either the recent msys2 update 
or new code inside x265-hg are the cause for 
"x265.exe" mis-compilation.

At this stage I am not convinced a suite update 
will cure me of this, but given the large number of 
commits since 09/03 and especially https://github.com/jb-alvarado/media-autobuild_suite/commit/ebce5aef56007002f6357e625b37abd02c044ee4, 
should I take some extra precautionary measures 
so as not to break the suite if I am advised to update it?

As ever, your patience with me is highly appreciated, 
your educated consult even more so!

**Comment by wiiaboo** [comment]
You should be always using the latest version of the suite to run things, otherwise you're trying to use patches that don't exist anymore and running into bugs that might have been fixed already.

I don't think it's a problem with the suite but with recent commits in x265, too.

FWIW, I can't reproduce that with my usual 10-bit only x265.exe.

Anyway, if you're not using the latest commit before creating the issue, I can't help. I try to keep the suite always relatively stable, but since files/patches are downloaded from master, the scripts have to be from master too or thing start breaking.

**Comment by Vangelis66** [comment]
I wrote:

> At this stage I am not convinced 
> a suite update will cure me of this  

@wiiaboo wrote:

> You should be always using the latest 
> version of the suite to run things
> (snip)
> if you're not using the latest commit 
> before creating the issue, I can't help

Although early morning in my timezone, 
I did manually update the suite to latest 
snapshot https://github.com/jb-alvarado/media-autobuild_suite/commit/ebce5ae.
As done previously,
_I deleted "/build/x265-hg", "/local32/bin-video/x265.exe",
"/local32/include/x265.h (& x265_config.h)",
"/local32/lib/pkgconfig/x265.pc" and "/local32/lib/libx265.a"
and opted to recompile only "kvazaar.exe" and "x265.exe"_.

Main suite batch file was re-run, no msys2/mingw32 
updates found, compilation proceeded straight to 
x265 now (since nothing else was updated in the meantime), 
with the same (as with older suite snapshot) "failure"; 
in the eyes of the suite, compilation was "successful", 
but end (unstripped) x265.exe binary won't run...

Manually composed [20160315044000Z_logs.zip](https://github.com/jb-alvarado/media-autobuild_suite/files/173283/20160315044000Z_logs.zip)
 

> FWIW, I can't reproduce that with my usual 10-bit only x265.exe.
> Edit: Even with multilib with all three, I can't reproduce here.

Have you tried to compile the x86 multilib [8bit+10bit+12bit] 
x265.exe binary with the msys32/mingw32 toolchain? 
(I think someone else in here is using WinXP SP3 32bit, 
maybe they should have a go?).

FWIW, the latest multilib binary compiled with the suite 
on this very machine was 
```
x265 [info]: HEVC encoder version 1.9+73-6d06de58c3163c19
x265 [info]: build info [Windows][GCC 5.3.0][32 bit] 8bit+10bit+12bit
x265 [info]: using cpu capabilities: MMX2 SSE2Fast SSSE3 Cache64 SlowShuffle
```
on Mar 8th 2016, 16:27:19 UTC

I'll be away from this laptop for the next 5-6 hours, 
in the meantime any more suggestions from 
whoever are welcome...

Regards.

**Comment by wiiaboo** [comment]
Try creating an issue in x265's bitbucket too. They're more likely to know what's wrong.
Canceling the suite (with Ctrl+c) some time after x265 has finished building so x265's logs are still available would be useful too.

**Comment by Vangelis66** [comment]
> Canceling the suite (with Ctrl+c) some time 
> after x265 has finished building 
> so x265's logs are still available 
> would be useful too.

Well, I decided to follow your tip, so 
1.I again removed every trace of anything 
x265 related inside suite's folders (as detailed 
previously in this thread) to force recompilation 
from a clean state...
2. I executed main suite batch file.

Updating the suite did find a number of updates, 
git was updated inside the minnty console and 
then that was auto-killed and in a win cmd window 
msys2-runtime and pacman received further updates, 
which were successfully installed. 
Then the suite, in a minnty console, proceeded to 
"Starting 32bit compilation of all tools", 
which I forcibly killed. 
 This was done to run first "autorebase.bat" 
(as admin), as described in https://github.com/jb-alvarado/media-autobuild_suite/issues/294#issuecomment-192938116, 
to be on the safe side...

After above procedure, I rerun the suite - 
compile.log is attached: [compile.log](https://github.com/jb-alvarado/media-autobuild_suite/files/174316/compile.log.txt)

As you can see, there is now no error produced by "strip.exe" 
and I am happy to report that compiled multilib binary runs OK:
```
G:\Compiler-v2\media-autobuild_suite\local32\bin-video>x265 -V
x265 [info]: HEVC encoder version 1.9+93-4a2f94a592511afa
x265 [info]: build info [Windows][GCC 5.3.0][32 bit] 8bit+10bit+12bit
x265 [info]: using cpu capabilities: MMX2 SSE2Fast SSSE3 Cache64 SlowShuffle
```
So, to sum up, the most likely culprit for my issue 
was not recent code inside x265-hg but the various 
msys2/mingw32 updates the suite applied in the early 
hours (UTC+0200) of Mon 14/03/2016; those must've 
crippled in some way the compiler as far as x265.exe 
was concerned; today's msys2-runtime (i686) updates 
have fixed - AFAICT - the MINGW32 compiler, 
so produced 32bit multilib x265.exe binary runs fine.

You can change issue's label to "not-our-bug" 
and close it afterwards, thank you very much...

**Comment by wiiaboo** [comment]
Weird. But that's the life with msys-dependant builds.
